### PR TITLE
[nocheck] Fix pdp dependency plot axes for categoricals

### PR DESCRIPTION
**Pull request by sebhrusen**
Without the fix, the PDP plots don't work with most versions of `matplotlib`

### CODE REVIEW

File: h2o-py/h2o/model/model_base.py [hunk 1]
```diff
@@ -1102,7 +1102,7 @@ def partial_plot(self, data, cols=None, destination_key=None, nbins=20, weight_c
         if cols is None and col_pairs_2dpdp is None:
             raise ValueError("Must specify either cols or col_pairs_2dpd to generate partial dependency plots.")
 
-        if col_pairs_2dpdp and targets and len(targets > 1):
+        if col_pairs_2dpdp and targets and len(targets) > 1:
```
Comment (sebhrusen) [inline h2o-py/h2o/model/model_base.py:1105]: also found this one

File: h2o-py/h2o/model/model_base.py [hunk 2]
```diff
@@ -1164,8 +1164,10 @@ def partial_plot(self, data, cols=None, destination_key=None, nbins=20, weight_c
         pps = json["partial_dependence_data"]
 
         # Plot partial dependence plots using matplotlib
-        return self.__generate_partial_plots(num_1dpdp, num_2dpdp, plot, server, pps, figsize, col_pairs_2dpdp, data, nbins,
-                                      kwargs["user_cols"], kwargs["num_user_splits"], plot_stddev, cols, save_plot_path, row_index, targets, include_na)
+        return self.__generate_partial_plots(num_1dpdp, num_2dpdp, plot, server, pps, figsize, 
+                                             col_pairs_2dpdp, data, nbins,
+                                             kwargs["user_cols"], kwargs["num_user_splits"], 
+                                             plot_stddev, cols, save_plot_path, row_index, targets, include_na)
```
Comment (sebhrusen) [inline h2o-py/h2o/model/model_base.py:1170]: formatting only

File: h2o-py/h2o/model/model_base.py [hunk 3]
```diff
@@ -1360,7 +1362,10 @@ def __set_axs_1d(self, axs, plot_stddev, cat, pp, col, row_index, target, includ
             fmt = 'o'
         else:
             fmt = '-'
-            axs.set_xlim(min(x), max(x))
+            if isinstance(x[0], str):
+                axs.set_xlim(0, len(x)-1)
+            else:
+                axs.set_xlim(min(x), max(x))
```
Comment (sebhrusen) [inline h2o-py/h2o/model/model_base.py:1368]: the fix to handle categoricals

Comment (wendycwong) [reply h2o-py/h2o/model/model_base.py:1368]: Great catch!

**Review by wendycwong — APPROVED** [review]
Looks great to me!

**Pull request merged**